Dockerfile

```
@@ -1,8 +1,9 @@
-FROM docker.mirror.hashicorp.services/node:14.17.0-alpine
+FROM docker.mirror.hashicorp.services/node:14.17.0-alpine AS base
 RUN apk add --update --no-cache git make g++ automake autoconf libtool nasm libpng-dev
 
-COPY ./package.json /website/package.json
-COPY ./package-lock.json /website/package-lock.json
+COPY ./package.json ./package-lock.json /website/
 WORKDIR /website
-RUN npm install -g npm@latest
-RUN npm install
\ No newline at end of file
+RUN npm install -g npm@latest && npm ci
+
+FROM base AS full
+COPY . /website
```

Loose understanding here:

I think you want something like 

```suggestion
FROM docker.mirror.hashicorp.services/node:14.17.0-alpine
COPY --from=base . /website
```

- Select a clean base image, like `node:14.17.0-alpine` again
- Copy stuff as needed from your _builder_ (named `base` here)
  - The exact `COPY` args here needs confirmation

---

I think we're gonna need the same dependencies as the base image (since we're running Next.js in dev mode), so at that point I think it makes more sense to use `FROM base` since that saves us from doing another `apk add`.

---

Oh that makes sense! (I was missing the intent behind the Dockerfile)